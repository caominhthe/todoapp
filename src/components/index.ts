export { default as TodoItem } from './TodoItem';
export { default as TodoInputGroup } from './TodoInputGroup';
