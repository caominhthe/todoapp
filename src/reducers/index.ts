export * from './authentication';
export * from './todo';
