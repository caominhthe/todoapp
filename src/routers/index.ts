export * from './types.d';
export { default as ScreenNames } from './Routes';
export { default as RootNavigator } from './rootNavigator';
export { default as Navigator } from './navigator';
