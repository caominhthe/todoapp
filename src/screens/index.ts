export * from './Authentication';
export * from './TodoScreen';
