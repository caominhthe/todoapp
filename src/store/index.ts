import { applyMiddleware, compose, createStore } from 'redux';

import AsyncStorage from '@react-native-async-storage/async-storage';
import persistCombineReducers from 'redux-persist/es/persistCombineReducers';
import persistStore from 'redux-persist/es/persistStore';
import rootReducer from './reducer';
import thunk from 'redux-thunk';

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  blacklist: ['authentication', 'todo'],
};

const persistedReducer = persistCombineReducers(persistConfig, rootReducer);

export const store = createStore(persistedReducer, applyMiddleware(thunk));

export const persistor = persistStore(store);
