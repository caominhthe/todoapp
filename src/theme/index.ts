export * from './pallete';
export * from './spacing';
export * from './typography';
