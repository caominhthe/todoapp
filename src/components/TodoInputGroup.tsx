import React from 'react';
import {
  Dimensions,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native';
import { spacing, typography } from '../theme';

interface TodoInputGroup {
  value?: string;
  onChangeText?: (text: string) => void;
  onPressButton?: () => void;
  button: React.ReactNode | any;
}

const TodoInputGroup: React.FC<TodoInputGroup> = ({
  value,
  onChangeText,
  onPressButton,
  button,
}) => {
  return (
    <View style={styles.textInputContainer}>
      <TextInput
        placeholder="Add new todo"
        onChangeText={onChangeText}
        style={styles.textInput}
        multiline
        value={value}
      />
      <TouchableOpacity testID="submit-button" onPress={onPressButton}>
        {button}
      </TouchableOpacity>
    </View>
  );
};

export default TodoInputGroup;

const { width: windowWidth } = Dimensions.get('window');

const styles = StyleSheet.create({
  textInputContainer: {
    flexDirection: 'row',
    alignItems: 'center',

    paddingHorizontal: spacing.medium,
    paddingBottom: spacing.tiny,
    width: windowWidth,
  },
  textInput: {
    flex: 1,
    fontSize: typography.fontSize.large,
    fontWeight: '700',
    flexShrink: 1,
  },
});
