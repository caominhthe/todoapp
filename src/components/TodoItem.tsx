import { StyleSheet, TouchableOpacity, View, Text } from 'react-native';
import React from 'react';
import { useCallback } from 'react';
import { Todo } from '../types';
import { spacing, typography } from '../theme';

interface TodoItemProps {
  todo: Todo;
  onDelete: (todo: Todo) => void;
  onPress: (todo: Todo) => void;
  isSelected: boolean;
}

const TodoItem: React.FC<TodoItemProps> = ({
  todo,
  onPress,
  onDelete,
  isSelected,
}) => {
  const onPressTodo = useCallback(() => {
    onPress?.(todo);
  }, [onPress, todo]);

  const onPressDelete = useCallback(() => {
    onDelete?.(todo);
  }, [onDelete, todo]);

  return (
    <View
      style={[styles.container, isSelected && styles.isSelected]}
      key={todo.id}
    >
      <TouchableOpacity
        testID="todo-item"
        style={styles.titleButton}
        onPress={onPressTodo}
      >
        <Text testID="item-title" style={styles.title}>
          {todo.title}
        </Text>
      </TouchableOpacity>
      <TouchableOpacity testID="delete-button" onPress={onPressDelete}>
        <Text style={styles.delete}>Delete</Text>
      </TouchableOpacity>
    </View>
  );
};

export default TodoItem;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    minHeight: 40,
    paddingVertical: spacing.small,
    borderWidth: 1,
    borderColor: 'grey',
    borderRadius: spacing.small,
    paddingHorizontal: spacing.large,
    paddingTop: spacing.small,
    marginTop: spacing.small,
    marginHorizontal: spacing.large,
  },
  delete: {
    color: 'blue',
    fontSize: typography.fontSize.medium,
  },
  rightIcon: {
    marginRight: spacing.medium,
    marginLeft: 'auto',
  },
  leftIcon: {
    marginLeft: spacing.medium,
  },
  title: {
    paddingHorizontal: spacing.medium,
    fontSize: typography.fontSize.medium,
    fontWeight: 'bold',
    flexWrap: 'wrap',
  },
  titleButton: {
    flex: 1,
  },
  isSelected: {
    backgroundColor: 'grey',
  },
});
