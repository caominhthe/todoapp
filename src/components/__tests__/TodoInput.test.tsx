import React from 'react';
import { Text } from 'react-native';
import { TodoInputGroup } from '@root/src/components';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';

describe('TodoInputGroup', () => {
  const onChangeText = jest.fn();
  const onSubmit = jest.fn();

  const button: React.ReactElement = <Text />;
  const tree = renderer.create(
    <TodoInputGroup
      value="test"
      onPressButton={onSubmit}
      onChangeText={onChangeText}
      button={button}
    />,
  );

  it('should render correctly', async () => {
    const treeJSON = tree.toJSON();
    expect(treeJSON).toMatchSnapshot();
  });

  it('has correct values', async () => {
    const { getByTestId } = render(
      <TodoInputGroup
        value="test"
        onPressButton={onSubmit}
        onChangeText={onChangeText}
        button={button}
      />,
    );
    const submitButton = getByTestId('submit-button');
    fireEvent.press(submitButton);
    expect(onSubmit).toBeCalled();
  });
});
