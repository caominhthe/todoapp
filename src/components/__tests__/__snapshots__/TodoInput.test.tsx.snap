// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInputGroup should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingBottom": 5,
      "paddingHorizontal": 16,
      "width": 750,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[MockFunction]}
    placeholder="Add new todo"
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
        "flexShrink": 1,
        "fontSize": 20,
        "fontWeight": "700",
      }
    }
    underlineColorAndroid="transparent"
    value="test"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="submit-button"
  >
    <Text />
  </View>
</View>
`;
